modules/data_release/jsx/uploadFileForm.js

```diff
@@ -0,0 +1,215 @@
+          onUserInput={this.updateFormElement}
+          required={true}
+          value={this.state.formData.file}
+        />
+        <TextboxElement
+          name='version'
+          label='Version'
+          onUserInput={this.updateFormElement}
+          required={false}
+          value={this.state.formData.version}
+        />
+        <ButtonElement label='Upload File'/>
+        <div className='row'>
+          <div className='col-sm-9 col-sm-offset-3'>
+            <ProgressBar value={this.state.uploadProgress}/>
+          </div>
+        </div>
+      </FormElement>
+    );
+  }
+
+  /**
+   * Update a form Element
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding element
+   */
+  updateFormElement(formElement, value) {
+    let formData = this.state.formData;
+
+    if (value === '') {
+      formData[formElement] = null;
+    } else {
+      formData[formElement] = value;
+    }
+
+    this.setState({formData: formData});
+  }
+
+  /**
+   * Validate and submit the upload
+   */
+  validateAndSubmit() {
+    let files = this.state.data.files ? this.state.data.files : [];
+    let formData = this.state.formData;
+
+    let errorMessage = {
+      Filename: undefined,
+      Filesize: undefined,
+    };
+
+    let hasError = {
+      Filename: undefined,
+      Filesize: undefined,
+    };
+
+    if (!formData.file) {
+      errorMessage.Filename = 'You must select a file to upload';
+      hasError.Filename = true;
+      this.setState({errorMessage, hasError});
+      return;
+    }
+
+    // Check that the size of the file is not bigger than the allowed size
+    let fileSize = formData.file ? Math.round((formData.file.size/1024)) : null;
+    const maxSizeAllowed = this.state.data.maxUploadSize;
+    if (parseInt(fileSize, 10) > parseInt(maxSizeAllowed, 10)*1024) {
+      let msg = 'File size exceeds the maximumn allowed (' + maxSizeAllowed + ')';
+      errorMessage['Filesize'] = msg;
+      hasError['Filesize'] = true;
+      swal({
+        title: 'Error',
+        text: msg,
+        type: 'error',
+        showCancelButton: true,
+      });
+      this.setState({errorMessage, hasError});
+      return;
+    }
+
+    // Grep the uploaded file name
+    let fileName = formData.file ? formData.file.name.replace(/\s+/g, '_') : null;
+
+    // Check for duplicate file names
+    let isDuplicate = files.indexOf(fileName);
+    if (isDuplicate >= 0) {
+      swal({
+        title: 'Are you sure?',
+        text: 'A file with this name already exists!\n Would you like to override existing file?',
+        type: 'warning',
+        showCancelButton: true,
+        confirmButtonText: 'Yes, I am sure!',
+        cancelButtonText: 'No, cancel it!',
+      }, function(isConfirm) {
+        if (isConfirm) {
+          this.uploadFile();
+        } else {
+          swal('Cancelled', 'Your imaginary file is safe :)', 'error');
```

Comment:
```suggestion
          swal('Cancelled', 'Your file is safe :)', 'error');
```

Reply:
But are my imaginary files in danger now? :(

Reply:
Not if you have imaginary backups